### PR TITLE
added Nefertari's backend query functions

### PR DESCRIPTION
**Pull request by Nefertari-Qin**

### CODE REVIEW

File: src/queries.js [hunk 1]
```diff
@@ -164,6 +164,100 @@ const deletePost = async (postId) => {
...
+        console.error(e);
+        throw e;
+    }
+};
+
+const getUpvotes = async (postId) => {
+    try {
+        const res = await db.query(`
+            SELECT SUM(U.upvotes)
+            FROM Rewards R, Award A, UpvoteRate U;
```
Comment (jacksonyuanjx) [inline src/queries.js:252]: You have an extra semi-colon here.

File: src/queries.js [hunk 2]
```diff
@@ -164,6 +164,100 @@ const deletePost = async (postId) => {
...
+        console.error(e);
+        throw e;
+    }
+};
+
+const getAwards = async () => {
+    try {
+        const { rows } = await db.query(`
+            SELECT awardName
+            FROM Award;
```
Comment (cheungis) [inline src/queries.js:239]: extra semicolon here

File: src/queries.js (hunk 1 above)
Comment (Nefertari-Qin) [reply src/queries.js:252]: right, thx, fixed

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,3161 @@
+{
```
[the remark this answers is not in the document]
Comment (Nefertari-Qin) [reply package-lock.json:1]: sure

File: src/queries.js [hunk 4]
```diff
@@ -248,6 +248,136 @@ const getPostsWithAllAwards = async () => {
 };
 
 // TODO: Nefertari's queries
+const editQuestionPost = async (content, postId) => {
+    try {
+        await db.query(`
```
Comment (jacksonyuanjx) [inline src/queries.js:253]: I think we also need a query to update the `mostRecentEditDate` field of the corresponding row in the `Post` table.

Comment (Nefertari-Qin) [reply src/queries.js]: thanks for catching that, fixed.

**Review by jacksonyuanjx — APPROVED** [review]

**Pull request merged**